src/librustdoc/clean/auto_trait.rs

```diff
@@ -168,7 +168,7 @@ fn clean_param_env<'tcx>(
 
     // FIXME(#111101): Incorporate the explicit predicates of the item here...
     let item_predicates: FxIndexSet<_> =
-        tcx.predicates_of(item_def_id).predicates.iter().map(|(pred, _)| pred).collect();
+        tcx.param_env(item_def_id).caller_bounds().iter().collect();
```

Comment:
This is basically a smol revert. Before by refactoring PR it looked very similar:

https://github.com/rust-lang/rust/blob/da01cced15b1a59e5eb40bdf5bb0be9a143d4e5a/src/librustdoc/clean/auto_trait.rs#L442